### ISSUE TITLE
discovery-ec2: 401 authentication failure on meta-data/placement/availability-zone http call

### ISSUE THREAD

**Issue by neuki** [elastic/elasticsearch#91585]
### Elasticsearch Version

Version: 7.17.7, Build: default/rpm/78dcaaa8cee33438b91eca7f5c7f56a70fec9e80/2022-10-17T15:29:54.167373105Z, JVM: 19

### Installed Plugins

discovery-ec2-7.17.7.jar

### Java Version

_bundled_

### OS Version

Linux ip-xx-xx-xx-xx.eu-central-1.compute.internal 4.14.296-222.539.amzn2.x86_64 #1 SMP Wed Oct 26 20:36:53 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

### Problem Description

If only the IMDSv2 mode is enabled, we are getting an 401 authentication failure:

> HTTP response code: 401 for URL: http://169.254.169.254/latest/meta-data/placement

Is the **meta-data-token missing** on this request?

See Ec2DiscoveryPlugin.getAvailabilityZoneNodeAttributes(...)

### Steps to Reproduce

- enable IMDSv2 on ec2 instance
   - aws ec2 modify-instance-metadata-options --instance-id xxxxxxxxxxxx --http-endpoint enabled --http-token required
- restart the elasticsearch server
- show logfile

### Logs (if relevant)

```
java.io.IOException: Server returned HTTP response code: 401 for URL: http://169.254.169.254/latest/meta-data/placement/availability-zone
        at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1993) ~[?:?]
        at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1585) ~[?:?]
        at java.security.AccessController.doPrivileged(AccessController.java:569) ~[?:?]
        at org.elasticsearch.discovery.ec2.SocketAccess.doPrivilegedIOException(SocketAccess.java:37) ~[discovery-ec2-7.17.7.jar:7.17.7]
        at org.elasticsearch.discovery.ec2.Ec2DiscoveryPlugin.getAvailabilityZoneNodeAttributes(Ec2DiscoveryPlugin.java:150) [discovery-ec2-7.17.7.jar:7.17.7]
        at org.elasticsearch.discovery.ec2.Ec2DiscoveryPlugin.additionalSettings(Ec2DiscoveryPlugin.java:125) [discovery-ec2-7.17.7.jar:7.17.7]
        at org.elasticsearch.plugins.PluginsService.updatedSettings(PluginsService.java:218) [elasticsearch-7.17.7.jar:7.17.7]
        at org.elasticsearch.node.Node.<init>(Node.java:414) [elasticsearch-7.17.7.jar:7.17.7]
        at org.elasticsearch.node.Node.<init>(Node.java:309) [elasticsearch-7.17.7.jar:7.17.7]
        at org.elasticsearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:234) [elasticsearch-7.17.7.jar:7.17.7]
        at org.elasticsearch.bootstrap.Bootstrap.setup(Bootstrap.java:234) [elasticsearch-7.17.7.jar:7.17.7]
        at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:434) [elasticsearch-7.17.7.jar:7.17.7]
        at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:169) [elasticsearch-7.17.7.jar:7.17.7]
        at org.elasticsearch.bootstrap.Elasticsearch.execute(Elasticsearch.java:160) [elasticsearch-7.17.7.jar:7.17.7]
        at org.elasticsearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:77) [elasticsearch-7.17.7.jar:7.17.7]
        at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:112) [elasticsearch-cli-7.17.7.jar:7.17.7]
        at org.elasticsearch.cli.Command.main(Command.java:77) [elasticsearch-cli-7.17.7.jar:7.17.7]
        at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:125) [elasticsearch-7.17.7.jar:7.17.7]
        at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:80) [elasticsearch-7.17.7.jar:7.17.7]
```

**Comment by neuki** [comment]
ping

**Comment by arteam** [comment]
Hey @neuki! The support for IMDSv2 was added in #84410 which was shipped in 8.2.0. Unfortunately, the 7.x branch at that time was already at the maintenance mode and only open to bugfixes, so #84410 wasn't backported to 7.17. 

We suggest you can upgrade to ES 8.2+ to pick this feature up if you can. Alternatively, you can try to use  [file-based seed hosts provider](https://www.elastic.co/guide/en/elasticsearch/reference/7.17/modules-discovery-hosts-providers.html#file-based-hosts-provider) and populate the file by querying the AWS API externally.

**Comment by neuki** [comment]
At the moment, we can not upgrade to 8.2, but the file based alternative works! 
Importent: If you use the workaround, you have also to set up the 'elastic zone awareness'. Set the following two properties in your elasitcsearch.yml or your cluster is no longer replicating shards:

node.attr.aws_availability_zone: {your zone e.g. eu-central-1a or eu-central-1b}
cluster.routing.allocation.awareness.attributes: aws_availability_zone
